scripts/utils/setup-roles.ts

```diff
@@ -0,0 +1,145 @@
+    }
+
+    if (FEE_CONTROLLER_ADDRESS) {
+        console.log("Fee controller address:", FEE_CONTROLLER_ADDRESS);
+    }
+
+    console.log(SECTION_SEPARATOR);
+
+    const loanCore = await ethers.getContractAt("LoanCore", LOAN_CORE_ADDRESS);
+    const vaultFactory = await ethers.getContractAt("VaultFactory", VAULT_FACTORY_ADDRESS);
+    const lenderNote = await ethers.getContractAt("PromissoryNote", LENDER_NOTE_ADDRESS);
+    const borrowerNote = await ethers.getContractAt("PromissoryNote", BORROWER_NOTE_ADDRESS);
+
+    // grant correct permissions for promissory note
+    // giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
+        await note.connect(admin).initialize(loanCore.address);
+    }
+
+    // grant LoanCore admin fee claimer permissions
+    const updateLoanCoreFeeClaimer = await loanCore.connect(admin).grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreFeeClaimer.wait();
+
+    console.log(`loanCore has granted fee claimer role: ${FEE_CLAIMER_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant LoanCore the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateLoanCoreAdmin = await loanCore.connect(admin).grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreAdmin.wait();
+
+    console.log(`loanCore has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant VaultFactory the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateVaultFactoryAdmin = await vaultFactory.connect(admin).grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateVaultFactoryAdmin.wait();
+
+    console.log(`vaultFactory has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant originationContoller the owner role to enable authorizeUpgrade onlyOwner
+    const updateOriginationControllerAdmin = await loanCore.connect(admin).grantRole(ORIGINATOR_ROLE, ORIGINATION_CONTROLLER_ADDRESS);
+    await updateOriginationControllerAdmin.wait();
+
+    console.log(`originationController has granted originator role: ${ORIGINATOR_ROLE} to address: ${ORIGINATION_CONTROLLER_ADDRESS}`);
+
+    // grant repaymentContoller the REPAYER_ROLE
+    const updateRepaymentControllerAdmin = await loanCore.connect(admin).grantRole(REPAYER_ROLE, REPAYMENT_CONTROLLER_ADDRESS);
+    await updateRepaymentControllerAdmin.wait();
+
+    console.log(`loanCore has granted repayer role: ${REPAYER_ROLE} to address: ${REPAYMENT_CONTROLLER_ADDRESS}`);
+    console.log(SECTION_SEPARATOR);
+
+    // renounce ownership from deployer
+    const renounceAdmin = await loanCore.connect(admin).renounceRole(ADMIN_ROLE, await deployer.getAddress());
+    await renounceAdmin.wait();
+
+    console.log(`loanCore has renounced admin role.`);
+
+    const renounceOriginationControllerAdmin = await loanCore.connect(admin).renounceRole(ORIGINATOR_ROLE, await deployer.getAddress());
```

Comment:
The `deployer` would never have had the `ORIGINATOR_ROLE`, but they would have had the `ADMIN_ROLE`, so they need to renounce that (after successfully granting it to `ADMIN_ADDRESS`, see above comment).